Framework/src/InfrastructureGenerator.cxx

```diff
@@ -800,4 +808,48 @@ void InfrastructureGenerator::generatePostProcessing(WorkflowSpec& workflow, con
   }
 }
 
+template <typename Type>
+auto createSinkInput(const std::string& detectorName, const std::string& name) -> framework::InputSpec
+{
+  const auto outputSpec = Type::createOutputSpec(detectorName, name);
+  auto input = DataSpecUtils::matchingInput(outputSpec);
+  input.binding = name;
+  return input;
+}
+
+void InfrastructureGenerator::generateBookkeepingQualitySink(WorkflowSpec& workflow, const InfrastructureSpec& infrastructureSpec)
```

Comment:
aside from that, you will also have to customize how this Data Processor treats inputs. By default, DPL will try to match the timeslice for all expected inputs, which makes sense for processing timeframe-based data, but since our messages are usually created based on timers, DPL will not be able to match them.

You can get inspired by `AggregatorRunnerFactory::customizeInfrastructure` and add basically the same thing for `BookkeepingQualitySink`. It will let it consume any incoming input immediately.